### PR TITLE
add kedge http client

### PR DESCRIPTION
**Pull request by mwitkow**
This adds an `http.Client` (in `kedge_client.New`) that will make smart decisions about dialing through a remote kedge or not, based on a `kedge_map`. Also added a Suffix `kedge_map` that expands to multiple clusters dynamically.

### CODE REVIEW

File: lib/map/iface.go [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
+var (
+	ErrNotKedgeDestination = errors.New("not a kedge destination")
+)
+
+// Mapper is an interface that allows you to direct traffic to different kedges.
+// These are used by client libraries.
 type Mapper interface {
 	// Map maps a target's DNS name (e.g. myservice.prod.ext.europe-cluster.local) to a (public) URL of the Kedge
 	// fronting that destination. The returned Scheme is deciding whether the Kedge connection is secure.
+	// An error c
```
Comment (improbable-yifan) [inline lib/map/iface.go]: finish your sentence pls

File: lib/map/suffix.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		return nil, errors.New("scheme must be http or https")
+	}
+	return &suffixMapper{
+		reverseMatchParts: reverse(strings.Split(matchPattern, ".")),
+		suffix:            suffix,
+		scheme:            scheme,
+	}, nil
+}
+
+func (s *suffixMapper) Map(targetAuthorityDnsName string) (*url.URL, error) {
```
Comment (improbable-yifan) [inline lib/map/suffix.go]: doc pls

File: lib/map/suffix.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+type suffixMapper struct {
+	reverseMatchParts []string
+
+	suffix string
+	scheme string
+}
+
+// Suffix is a `kedge.Mapper` that copies over wildcarded parts of a URL.
+//
+// For example: given a matchPattern of *.*.clusters.local, and a URL of `myservice.mynamespace.svc.us1.prod.clusters.local`
```
Comment (improbable-yifan) [inline lib/map/suffix.go]: did you mean to put `*.clusters.local` for your pattern?

Comment (mwitkow) [inline lib/map/suffix.go]: no, see tests.

File: lib/map/suffix.go (hunk 2 above)
Comment (mwitkow) [inline lib/map/suffix.go]: it's an implementation of an interface that has docs

File: lib/map/suffix.go (hunk 3 above)
Comment (improbable-yifan) [inline lib/map/suffix.go]: oh, ignore me I miss read the URL as `us1-prod` instead of `us1.prod`

**Pull request merged**